docs/apache-airflow/core-concepts/tasks.rst

```diff
@@ -243,9 +243,103 @@ Zombie/Undead Tasks
 
 No system runs perfectly, and task instances are expected to die once in a while. Airflow detects two kinds of task/process mismatch:
 
-* *Zombie tasks* are tasks that are supposed to be running but suddenly died (e.g. their process was killed, or the machine died). Airflow will find these periodically, clean them up, and either fail or retry the task depending on its settings.
+* *Zombie tasks* are ``TaskInstances`` stuck in a ``running`` state despite their associated jobs being inactive
+  (e.g. their process didn't send a recent heartbeat as it got killed, or the machine died). Airflow will find these
+  periodically, clean them up, and either fail or retry the task depending on its settings.
+
+* *Undead tasks* are tasks that are *not* supposed to be running but are, often caused when you manually edit Task
+  Instances via the UI. Airflow will find them periodically and terminate them.
+
+
+Below is the code snippet from the Airflow scheduler that runs periodically to detect zombie/undead tasks.
+
+.. exampleinclude:: /../../airflow/jobs/scheduler_job_runner.py
+    :language: python
+    :start-after: [START find_zombies]
+    :end-before: [END find_zombies]
+
+
+The explanation of the criteria used in the above snippet to detect zombie tasks is as below:
```

Comment:
We shouldn't go into details explaining the code. I'm okay with pointing out the code location but not with the code explanation because the code can change at any time and we might not remember to update this explanation. Similarly, the reproduction steps for zombie tasks are not necessary for a concept doc IMO.

Reply:
okay. I initially wrote in the ticket description that it would nice to explain how exactly zombies are detected and also steps for reproducing them https://github.com/apache/airflow/issues/35698
and hence added here thinking it might be helpful :)

Reply:
Yeah agree with @ephraimbuddy that showing airflow code internals in this case is a bit too much . This description is really valuable because it explains WHT happens and WHEN (and this part is cool) . But showing the details is not very useful. Users always can go and check it and when it will change it will get confusing a bit for them I think.

I'd just remove the implementation and leave all the rest